doc/api/fs.md

```diff
@@ -6546,6 +6582,39 @@ added:
 
 The base path that this {fs.Dirent} object refers to.
 
+### Class: `fs.FilterOutput`
+
+A {fs.FilterOutput} object provides identifiers for file system artifacts. This
+object is passed into the function provided to option `recursiveFilter` of
+[`fsPromises.readdir()`][], [`fs.readdir()`][], and [`fs.readdirSync()`][] when
+option `recursive` is assigned `true`.
+
+```console
+FilterOutput {
+  depth: 2,
+  itemName: 'fileName.ts',
+  itemPath: '/users/userAccount/notes/fileName.ts',
+  parentName: 'notes',
+  parentPath: '/users/userAccount/notes'
+}
+```
```

Comment:
Most of this seems pretty redundant. For consistency with `fs.cp()`, why not just pass the (potentially relative) path to the function? Determining the depth is simple enough if necessary, and so is determining the `parentName` etc.